DA-175: Fix backend tests

judembo opened the pull request:
Previously, backend tests weren't running anymore because the strategy
for generating PK's hat been changed from GenerationType.AUTO to
GenerationType.IDENTITY. With identity sequencing id's are not
assigned before flush() or commit() is called, so a call of em.flush()
was added after each call of persist.
Additionally, some tests weren't running because they were trying to catch
the wrong Exception types.

JeannedArk commented on src/test/java/de/unisaarland/discanno/test/ProjectFacadeRESTTest.java:
```diff
@@ -39,19 +39,15 @@ public void configureRESTFacade() {
 
         Users admin = TestDataProvider.getAdmin();
         admin.setSession(BaseTest.SESSION_STR);
         em.persist(admin);
```
It's better to create an extra method for this operation, because always calling
em.persist(...)
em.flush()
creates a lot of code duplication

JeannedArk commented on src/test/java/de/unisaarland/discanno/test/SchemeTest.java:
```diff
@@ -32,19 +32,15 @@
     public void configService() {
         super.configureService(service);
     }
+
```
Delete this blank line

JeannedArk commented on src/test/java/de/unisaarland/discanno/test/ProjectFacadeRESTTest.java:
```diff
@@ -39,19 +39,15 @@ public void configureRESTFacade() {
         super.configureService(schemeRESTService);
         super.configureService(documentRESTService);
     }
+
```
Delete this blank line

JeannedArk commented:
With every test run a file "Project1_Politics1_admin@web.de.xml" and "discanno_Project1.zip" are created. Create an @after annotated method in the corresponding class where these files are cleaned up again.

judembo committed:
DA-175: Fix backend tests

Previously, backend tests weren't running anymore because the strategy
for generating PK's hat been changed from GenerationType.AUTO to
GenerationType.IDENTITY. With identity sequencing id's are not
assigned before flush() or commit() is called.
Additionally, some exception types were corrected, a method for cleaning
up files generated during testing was added, and the test for exporting
projects was extended.
